lib/src/main/kotlin/com/swmansion/starknet/data/Cairo1ClassHashCalculator.kt

```diff
@@ -6,9 +6,11 @@ import com.swmansion.starknet.data.types.*
 import kotlinx.serialization.json.Json
 
 object Cairo1ClassHashCalculator {
+    private val jsonWithIgnoreUnknownKeys by lazy { Json { ignoreUnknownKeys = true } }
 
     fun computeSierraClassHash(contract: Cairo1ContractDefinition): Felt {
-        val contractClass = Json.decodeFromJsonElement(ContractClass.serializer(), contract.toJson())
+        val json = if (contract.ignoreUnknownJsonKeys) jsonWithIgnoreUnknownKeys else Json
+        val contractClass = json.decodeFromJsonElement(ContractClass.serializer(), contract.toJson())
```

Comment:
Do we want to additionally test this? We would have to probably keep altered compiled codes of Cairo 1 contracts, and we previously tried to avoid that.

Reply:
nah, should be fine without explicit test